app.js

```diff
@@ -1,6 +1,107 @@
 const express = require('express')
+const app = express();
 
-const app = express()
+const Event = require('./Event').Event;
+const bodyParser = require('body-parser');
+const eventsArray = new Array();
+/*class EventManager {
+	constructor() {*/
+
+		eventsArray.push(new Event(1, 'Conferencia', 'Node.js, Pair Programming', '2017-03-15'));
+		eventsArray.push(new Event(2, 'Concierto', 'Sinfonica', '2017-03-11'));
+		eventsArray.push(new Event(3, 'Cita Dentista', 'Limpieza Dental', '2017-03-09'));
+		eventsArray.push(new Event(4, 'Clases de frances', 'Iniciando', '2017-03-01'));
+		eventsArray.push(new Event(5, 'Cena', 'Conunidad Agile', '2017-02-15'));
+	/*}*/
+
+	function getAll(){
+		return new Promise( function (resolve, reject) {
+			try{
+				resolve(eventsArray);
+			}
+			catch(ex){
+				reject(ex);
+			}
+		});
+	};
+
+	function getEventById(id){
+		return new Promise( function (resolve, reject) {
+			try{
+				let eventoEncontrado = eventsArray.find(evento => evento.id == id);
+				resolve(eventoEncontrado);
+			}
+			catch(ex){
+				reject(ex);
+			}
+		});
+	}
+
+	function createEvent(id,title,description,date){
+		return new Promise( function (resolve, reject) {
+			try{
+				let idEncontrado = eventsArray.find(evento => evento.id == id);
+				if(idEncontrado == undefined){
+					let newEvent = new Event(id, title, description, date);
+					eventsArray.push(newEvent);
+					resolve(newEvent);
+				}else{
+					resolve(undefined);
```

Comment:
I think that a better option is to emit an error (reject) about the duplicated id